packages/server/graphql/types/AzureDevOpsIntegration.ts

```diff
@@ -0,0 +1,158 @@
+  instanceIds: string[]
+}
+
+interface AzureDevOpsAuth
+  extends Omit<IGetAzureDevOpsAuthByUserIdTeamIdQueryResult, 'azureDevOpsSearchQueries'> {
+  azureDevOpsSearchQueries: {
+    id: string
+    queryString: string
+    projectKeyFilters?: string[]
+    lastUsedAt: Date
+    isWIQL: boolean
+  }[]
+}
+
+const AzureDevOpsIntegration = new GraphQLObjectType<any, GQLContext>({
+  name: 'AzureDevOpsIntegration',
+  description: 'The Azure DevOps auth + integration helpers for a specific team member',
+  fields: () => ({
+    auth: {
+      description: 'The OAuth2 Authorization for this team member',
+      type: TeamMemberIntegrationAuthOAuth2,
+      resolve: async (
+        {teamId, userId}: {teamId: string; userId: string},
+        _args: unknown,
+        {dataLoader}
+      ) => {
+        return dataLoader
+          .get('teamMemberIntegrationAuths')
+          .load({service: 'azureDevOps', teamId, userId})
+      }
+    },
+    id: {
+      type: new GraphQLNonNull(GraphQLID),
+      description: 'Composite key in ado:teamId:userId format',
+      resolve: ({teamId, userId}: {teamId: string; userId: string}) => `ado:${teamId}:${userId}`
+    },
+    isActive: {
+      description: 'true if the auth is valid, else false',
+      type: new GraphQLNonNull(GraphQLBoolean),
+      resolve: ({accessToken}) => !!accessToken
+    },
+    accessToken: {
+      description:
+        'The access token to Azure DevOps. null if no access token available or the viewer is not the user',
+      type: GraphQLID
+      // Add resolver
+    },
+    accountId: {
+      type: new GraphQLNonNull(GraphQLID),
+      description: 'The Azure DevOps account ID'
+    },
+    instanceIds: {
+      type: new GraphQLNonNull(new GraphQLList(new GraphQLNonNull(GraphQLID))),
+      description: 'The Azure DevOps instance IDs that the user has granted'
+    },
+    createdAt: {
+      type: new GraphQLNonNull(GraphQLISO8601Type),
+      description: 'The timestamp the provider was created'
+    },
+    teamId: {
+      type: new GraphQLNonNull(GraphQLID),
+      description: 'The team that the token is linked to'
+    },
+    updatedAt: {
+      type: new GraphQLNonNull(GraphQLISO8601Type),
+      description: 'The timestamp the token was updated at'
+    },
+    userId: {
+      type: new GraphQLNonNull(GraphQLID),
+      description: 'The user that the access token is attached to'
+    },
+    workItems: {
+      type: new GraphQLNonNull(new GraphQLList(new GraphQLNonNull(AzureDevOpsWorkItem))),
+      description:
+        'A list of work items coming straight from the azure dev ops integration for a specific team member',
+      resolve: async (
+        {teamId, userId}: AzureDevOpsAuth,
+        {},
+        {authToken, dataLoader}: GQLContext
+      ) => {
+        const viewerId = getUserId(authToken)
+        if (viewerId !== userId) {
+          const err = new Error('Cannot access another team members issues')
```

Comment:
+1 usually we would check for team membership

Reply:
This change has been made